src/main/java/com/nickdsantos/onedrive4j/OneDrive.java

```diff
@@ -168,15 +168,21 @@ public AccessToken getAccessTokenFromRefreshToken(String refreshToken) throws IO
 			HttpPost httpPost = new HttpPost(uri);
 			httpPost.setEntity(formEntity);
 
-			Map<Object, Object> rawToken = httpClient.execute(httpPost, new OneDriveJsonToMapResponseHandler());
-			if (rawToken != null) {
+			Map<Object, Object> rawResponse = httpClient.execute(httpPost, new OneDriveJsonToMapResponseHandler());
+
+			if (rawResponse.containsKey("error")) {
```

Comment:
Indeed. I didn't get a warning about this in IDEA, so maybe it thought the return value couldn't be null.

Or maybe it just means not enough warnings are enabled yet.